test/unit/test_multipart.py

```diff
@@ -192,35 +194,54 @@ class TestStreamFileMixin(unittest.TestCase):
 	test__gen_file_end -- test the _gen_file_end function against example output
 	"""
 
-	def test__gen_file(self):
+	def do_test__gen_file(self, name, file_location, abspath):
 		"""Test the _gen_file function against sample output."""
-		name = "functional/fake_dir/fsdfgh"
 		generator = StreamFileMixinSub(name)
-
 		file = io.BytesIO()
 		file.write(b"!234")
 		file.seek(0)
 
-		expected = b'--' + generator._boundary.encode() + b'\r\nContent-Disposition: file; '\
-		         + b'filename="functional%2Ffake_dir%2Ffsdfgh"\r\nContent-Type: '\
-		         + b'text/plain\r\n\r\n' \
+		expected = b'--' + generator._boundary.encode() + b'\r\n'
+		expected += b'Abspath: ' + name.encode(ENC) + b'\r\n' if abspath else b''
+		expected += b'Content-Disposition: file; '\
+		         + b'filename="' + urllib.parse.quote_plus(name).encode(ENC) + b'"\r\n'\
```

Comment:
Replace `urllib.parse.quote_plus(name).encode(ENC)` by `urllib_parse.quote_plus(name).encode(ENC)` and add the import `from six.moves import urllib_parse` to the top of the file to make the Python 2.7 error go away.

Reply:
Fixed.